runtime/builtin_types.go

```diff
@@ -631,15 +631,39 @@ func builtinRound(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 			return nil, raised
 		}
 	}
-	// TODO: implement this.
-	if ndigits != 0 {
-		return nil, f.RaiseType(NotImplementedErrorType, "round with ndigits is not implemented in grumpy")
-	}
 	number, isFloat := floatCoerce(args[0])
+
 	if !isFloat {
 		return nil, f.RaiseType(TypeErrorType, "a float is required")
 	}
-	return NewFloat(math.Floor(number + 0.5)).ToObject(), nil
+
+	if math.IsNaN(number) || math.IsInf(number, 0) || number == 0.0 {
+		return NewFloat(number).ToObject(), nil
+	}
+
+	var result float64
+	if ndigits == 0 {
+		result = round(number, ndigits)
+		// round(12.5) => 13, round(-12.5) => -13
```

Comment:
````
/**
In Python 2, ndigits <0 treats the decimal point as .0, and the integer as | Ndigits | It is masked to 0 as | Digits |.
To do this, divide -12.5 by pow (10, | ndigits |) (this will be -1.25) and round it. Finally, multiplying by pow (10, | ndigits |) yields -10.0.
**/

````
@trotterdylan 
Will be a good enough?